www/apps/resources/app/recipes/marketplace/examples/vendors/page.mdx

```diff
@@ -0,0 +1,1344 @@
+  })
+})
+
+export default VendorAdmin
+```
+
+This creates a `VendorAdmin` data model, which represents an admin of a vendor.
+
+Then, add a migration file at `src/modules/marketplace/migrations/Migration20240708151444.ts`:
+
+```ts title="src/modules/marketplace/migrations/Migration20240708151444.ts"
+import { Migration } from '@mikro-orm/migrations';
+
+export class Migration20240708151444 extends Migration {
+
+  async up(): Promise<void> {
+    this.addSql('create table if not exists "vendor" ("id" text not null, "handle" text not null, "name" text not null, "logo" text null, "created_at" timestamptz not null default now(), "updated_at" timestamptz not null default now(), "deleted_at" timestamptz null, constraint "vendor_pkey" primary key ("id"));');
+    this.addSql('CREATE UNIQUE INDEX IF NOT EXISTS "IDX_vendor_handle_unique" ON "vendor" (handle) WHERE deleted_at IS NULL;');
+
+    this.addSql('create table if not exists "vendor_admin" ("id" text not null, "first_name" text null, "last_name" text null, "email" text not null, "vendor_id" text not null, "created_at" timestamptz not null default now(), "updated_at" timestamptz not null default now(), "deleted_at" timestamptz null, constraint "vendor_admin_pkey" primary key ("id"));');
+    this.addSql('CREATE UNIQUE INDEX IF NOT EXISTS "IDX_vendor_admin_email_unique" ON "vendor_admin" (email) WHERE deleted_at IS NULL;');
+    this.addSql('CREATE INDEX IF NOT EXISTS "IDX_vendor_admin_vendor_id" ON "vendor_admin" (vendor_id) WHERE deleted_at IS NULL;');
+
+    this.addSql('alter table if exists "vendor_admin" add constraint "vendor_admin_vendor_id_foreign" foreign key ("vendor_id") references "vendor" ("id") on update cascade;');
+  }
+
+  async down(): Promise<void> {
+    this.addSql('alter table if exists "vendor_admin" drop constraint if exists "vendor_admin_vendor_id_foreign";');
+
+    this.addSql('drop table if exists "vendor" cascade;');
+
+    this.addSql('drop table if exists "vendor_admin" cascade;');
+  }
+
+}
+```
+
+Next, create the main service of the module at `src/modules/marketplace/service.ts` with the following content:
+
+export const serviceHighlights = [
+  ["11", "createVendors", "Override the `createVendors` method generated by `MedusaService`."]
+]
+
+```ts title="src/modules/marketplace/service.ts" highlights={serviceHighlights}
+import { MedusaService } from "@medusajs/utils"
+import Vendor from "./models/vendor"
+import VendorAdmin from "./models/vendor-admin"
+import { CreateVendorData, VendorData } from "./types"
+
+class MarketplaceModuleService extends MedusaService({
+  Vendor,
+  VendorAdmin
+}) {
+  // @ts-expect-error
+  async createVendors(
+    data: CreateVendorData[]
+  ): Promise<VendorData[] | VendorData> {
+    const input = (Array.isArray(data) ? data : [data]).map(
+      (vendor) => {
+        if (!vendor.handle) {
+          vendor.handle = vendor.name.toLowerCase().replace(/\s/g, "-")
+        }
+
+        return vendor
+      }
+    )
```

Comment:
thought: I understand the idea with the `handle` formatting, but for the sake of the recipe, wouldn't it be cooler to have an empty service? This would make it more obvious that CRUD is coming out of the box.